--- origsrc/gconf-editor-2.26.0/configure.in	2009-03-17 09:30:40.000000000 -0500
+++ src/gconf-editor-2.26.0/configure.in	2009-04-19 18:21:07.343125000 -0500
@@ -35,9 +35,16 @@ changequote([,])dnl
 PKG_CHECK_MODULES(GCONF_EDITOR,
 		  gconf-2.0 >= 2.9.2
 		  gtk+-2.0 >= 2.12.0
-		  polkit-dbus >= 0.7
 		  dbus-glib-1 >= 0.71)
 
+PKG_CHECK_MODULES(POLKIT_DBUS,
+		  polkit-dbus >= 0.7,
+		  have_polkit_dbus=yes, have_polkit_dbus=no)
+
+if test "$have_polkit_dbus" = "yes"; then
+	AC_DEFINE(HAVE_POLKIT_DBUS, [], [Define if polkit-dbus is available])
+fi
+
 if test "$enable_maintainer_mode" = "yes"; then
 	DISABLE_DEPRECATED_CFLAGS="-DG_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED \
 -DG_DISABLE_SINGLE_INCLUDES -DGTK_DISABLE_SINGLE_INCLUDES"
--- origsrc/gconf-editor-2.26.0/src/Makefile.am	2009-04-19 21:13:44.093125000 -0500
+++ src/gconf-editor-2.26.0/src/Makefile.am	2009-04-19 21:39:02.671250000 -0500
@@ -3,7 +3,8 @@ INCLUDES = \
 	-DDATADIR=\""$(pkgdatadir)"\" 	\
 	-DICONDIR="\"$(pkgdatadir)/icons\"" \
 	$(DISABLE_DEPRECATED_CFLAGS) \
-	$(GCONF_EDITOR_CFLAGS)
+	$(GCONF_EDITOR_CFLAGS) \
+	$(POLKIT_DBUS_CFLAGS)
 
 bin_PROGRAMS = gconf-editor
 
@@ -54,7 +55,8 @@ regenerate-built-sources:
 
 
 gconf_editor_LDADD = \
-	$(GCONF_EDITOR_LIBS)
+	$(GCONF_EDITOR_LIBS) \
+	$(POLKIT_DBUS_LIBS)
 
 EXTRA_DIST = \
 	gconf-marshal.list
--- origsrc/gconf-editor-2.26.0/src/gconf-editor-window.c	2009-03-17 09:29:30.000000000 -0500
+++ src/gconf-editor-2.26.0/src/gconf-editor-window.c	2009-04-19 18:40:53.452500000 -0500
@@ -688,10 +688,12 @@ gconf_editor_popup_window_set_as_default
 
   if (can_use_pk) {
     gconf_client_suggest_sync (gconfwindow->client, NULL);
+#ifdef HAVE_POLKIT_DBUS
     gconf_pk_set_default_async (path,
                                 GDK_WINDOW_XWINDOW (GTK_WIDGET (window)->window),
                                 (GFunc) gconf_editor_popup_policykit_callback,
                                 g_object_ref (window), g_object_unref);
+#endif
   }
 }
 
@@ -736,10 +738,12 @@ gconf_editor_popup_window_set_as_mandato
 
   if (can_use_pk) {
     gconf_client_suggest_sync (gconfwindow->client, NULL);
+#ifdef HAVE_POLKIT_DBUS
     gconf_pk_set_mandatory_async (path,
                                   GDK_WINDOW_XWINDOW (GTK_WIDGET (window)->window),
                                   (GFunc) gconf_editor_popup_policykit_callback,
                                   g_object_ref (window), g_object_unref);
+#endif
   }
 }
 
@@ -934,10 +938,18 @@ list_view_button_press_event (GtkTreeVie
 						  window->type != GCONF_EDITOR_WINDOW_TYPE_DEFAULTS);
 			gtk_widget_set_sensitive (gtk_ui_manager_get_widget (window->ui_manager, "/GConfKeyPopupMenu/DefaultKey"), 
 						  (gconf_util_can_edit_defaults () && window->type != GCONF_EDITOR_WINDOW_TYPE_DEFAULTS) ||
+#ifdef HAVE_POLKIT_DBUS
 						  (gconf_pk_can_set_default () && window->type == GCONF_EDITOR_WINDOW_TYPE_NORMAL));
+#else
+                          FALSE);
+#endif
 			gtk_widget_set_sensitive (gtk_ui_manager_get_widget (window->ui_manager, "/GConfKeyPopupMenu/MandatoryKey"), 
 						  (gconf_util_can_edit_mandatory () && window->type != GCONF_EDITOR_WINDOW_TYPE_MANDATORY) ||
+#ifdef HAVE_POLKIT_DBUS
 						  (gconf_pk_can_set_mandatory () && window->type == GCONF_EDITOR_WINDOW_TYPE_NORMAL));
+#else
+                          FALSE);
+#endif
 			
 			gtk_tree_path_free (path);
 		}
@@ -1037,10 +1049,18 @@ gconf_editor_window_list_view_popup_menu
 					  window->type != GCONF_EDITOR_WINDOW_TYPE_DEFAULTS);
 		gtk_widget_set_sensitive (gtk_ui_manager_get_widget (window->ui_manager, "/GConfKeyPopupMenu/DefaultKey"), 
 					  (gconf_util_can_edit_defaults () && window->type != GCONF_EDITOR_WINDOW_TYPE_DEFAULTS) ||
+#ifdef HAVE_POLKIT_DBUS
 					  (gconf_pk_can_set_default () && window->type == GCONF_EDITOR_WINDOW_TYPE_NORMAL));
+#else
+					  FALSE);
+#endif
 		gtk_widget_set_sensitive (gtk_ui_manager_get_widget (window->ui_manager, "/GConfKeyPopupMenu/MandatoryKey"), 
 					  (gconf_util_can_edit_mandatory () && window->type != GCONF_EDITOR_WINDOW_TYPE_MANDATORY) ||
+#ifdef HAVE_POLKIT_DBUS
 					  (gconf_pk_can_set_mandatory () && window->type == GCONF_EDITOR_WINDOW_TYPE_NORMAL));
+#else
+					  FALSE);
+#endif
 			
 	}
 	else {
--- origsrc/gconf-editor-2.26.0/src/gconf-policykit.c	2009-03-17 09:29:30.000000000 -0500
+++ src/gconf-editor-2.26.0/src/gconf-policykit.c	2009-04-19 18:05:02.561875000 -0500
@@ -26,6 +26,8 @@
 #  include "config.h"
 #endif
 
+#ifdef HAVE_POLKIT_DBUS
+
 #include <stdlib.h>
 #include <stdio.h>
 #include <fcntl.h>
@@ -434,3 +436,5 @@ gconf_pk_set_mandatory_async (const gcha
         set_key_async (data);
         _gconf_pk_data_unref (data);
 }
+
+#endif /* HAVE_POLKIT_DBUS */
